### ISSUE TITLE
Contribution Request | MacOS Support

### ISSUE THREAD

**Issue by albe-rosado** [creating2morrow/neveko#35]
Would you welcome a PR to add support for MacOS?

**Comment by albe-rosado** [comment]
I just took a quick look, and besides hardcoded paths and variable names, `i2p-zero` is the hardest part, I can work on that too, in the repo you already cloned. Dunno if is something you are open to support.

**Comment by creating2morrow** [comment]
@albe-rosado yes that would be helpful. Lmk what you need on my end. I don't have a Mac to verify anything so if you could get another user to check that would be great. Thanks.

**Comment by albe-rosado** [comment]
Nothing so far, I will start working on it soon. Thanks! Is there any place online you hang out, in case we want to discuss stuff? Thanks

**Comment by creating2morrow** [comment]
>  Is there any place online you hang out, in case we want to discuss stuff?

No, I try to limit my exposure to the clear as much as possible. There is a monero ecosystem matrix channel I think. There is a nevekonet i2p irc server address in the i2p release server home page. http://c2m66oddrzozztxyzjegbdwtgbeiibq5vz2tpchmqamrzcahcfoq.b32.i2p/index.txt

I can join the i2p irc server there at 1200UTC daily and check messages. I will also check the GH daily if I have time.

**Comment by albe-rosado** [comment]
Just wanted to touch base, I have been pretty busy recently, but I have been working on [Rust bindings for i2pd](https://git.alberosado.me/albe/i2pd-sys) , the reason I went with `i2pd` is because is pretty fast, and is very easy to integrate with, as is just a portable binary supported on all major platforms and the team behind seems to be very active. i2p-zero in MacOS relies on an older version of JDK, I'm having a hard time even downloading, also it seems to not have a lot of work recently (maybe is defunct) .  Next I will remove the hardcoded paths in the bindings package and have a POC of neveko using such bindings, which in my opinion would be great as we wont have to execute binaries from code. Regarding `monerod`, I will take a look to see if there is any Rust implementation or existing bindings. I will cut a CR once I have i2pd integrated. I'm open to any other idea.

**Comment by creating2morrow** [comment]
@albe-rosado there has been mention of i2pd in other issues. The Rust bindings for i2pd would be a great and welcome addition. As for Rust bindings for Monero, I'm not aware of any. In the future it might be worth considering [Cuprate](https://github.com/Cuprate/cuprate) integration, but that wouldn't be anytime soon. Thanks for the effort. Look forward to the POC.
